enterprise/backend/src/metabase_enterprise/advanced_permissions/query_processor/middleware/permissions.clj

```diff
@@ -32,16 +35,23 @@
   ;; Remove the :native key (containing the transpiled MBQL) so that this helper function doesn't think the query is
   ;; a native query. Actual native queries are dispatched to a different method by the :type key.
   (let [{:keys [table-ids card-ids native?]} (query-perms/query->source-ids (dissoc query :native))
-        table-perms (if (or native? (seq card-ids))
+        table-perms (if native?
```

Comment:
the comment on 35 reads that this query cannot be native, should it be reworded? to something like:

```
  ;; Remove the :native key (containing the transpiled MBQL) so that this helper function doesn't mistakenly classify
  ;; the query as a native query. Actual native queries are dispatched to a different method by the :type key.
  ```
  
Actually, I don't see why the dissoc matters since a different defmethod will handle :native. If it's clear to you tho please update the comment.

Reply:
Good point, the dissoc is unnecessary now, I've removed it